app/src/main/java/org/mozilla/tv/firefox/ScreenController.kt

```diff
@@ -212,6 +213,7 @@ class ScreenController {
                     .hide(fragmentManager.pocketFragment())
                     .hide(fragmentManager.settingsFragment())
                     .commitNow()
+                fragmentManager.webRenderFragment().onShowEngineView()
```

Comment:
We're adding `onShowEngineView` each time currentActiveScreen is assigned to WEB_RENDER: this approach is fragile because if the code changes and we have a new assignment, we may forget to add this line.

What if we were reactive and did it as an observer instead? Do the LiveData observers get called before the View focus changes? If not, could we write a small wrapper around the currentActiveScreen update so that our observer can get called first before the LiveData changes?

Reply:
That's a good point.  The LiveData observers are called synchronously when the value is set, and asynchronously when post is used.  Changing these to set the value before kicking off the transaction would be trivial, but I'll have to double check that it won't have negative effects.

Good catch.